pkg/operator/sync.go

```diff
@@ -244,6 +246,93 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
 		return fmt.Errorf("refusing to read images.json version %q, operator version %q", imgs.ReleaseVersion, optrVersion)
 	}
 
+	// handle image registry certificates.
+	// parse these, add them to ctrlcfgspec and then handle these in the daemon write to disk function
+	cfg, err := optr.imgLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+	imgRegistryUsrData := []v1.ImageRegistryBundle{}
+	if cfg.Spec.AdditionalTrustedCA.Name != "" {
+		cm, err := optr.clusterCmLister.ConfigMaps("openshift-config").Get(cfg.Spec.AdditionalTrustedCA.Name)
+		if err != nil {
+			klog.Warningf("could not find configmap specified in image.config.openshift.io/cluster with the name %s", cfg.Spec.AdditionalTrustedCA.Name)
+		} else {
+			for key, d := range cm.Data {
+				raw, err := base64.StdEncoding.DecodeString(d)
+				if err != nil {
+					imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+						File: key,
+						Data: []byte(d),
+					})
+				} else {
+					imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+						File: key,
+						Data: raw,
+					})
+				}
+			}
+			for key, bd := range cm.BinaryData {
+				imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+					File: key,
+					Data: bd,
+				})
+			}
+		}
+	}
+
+	imgRegistryData := []v1.ImageRegistryBundle{}
+	cm, err := optr.clusterCmLister.ConfigMaps("openshift-config-managed").Get("image-registry-ca")
+	if err != nil {
+		klog.Info("the image registry is not installed so the MCO will not distribute its certificates.")
+	} else {
+		for key, d := range cm.Data {
+			raw, err := base64.StdEncoding.DecodeString(d)
+			if err != nil {
+				imgRegistryData = append(imgRegistryData, v1.ImageRegistryBundle{
+					File: key,
+					Data: []byte(d),
+				})
+			} else {
+				imgRegistryData = append(imgRegistryData, v1.ImageRegistryBundle{
+					File: key,
+					Data: raw,
+				})
+			}
+		}
+		for key, bd := range cm.BinaryData {
+			imgRegistryData = append(imgRegistryData, v1.ImageRegistryBundle{
+				File: key,
+				Data: bd,
+			})
+		}
+	}
+
+	mergedData := append(imgRegistryData, imgRegistryUsrData...)
+	caData := make(map[string]string, len(mergedData))
+	for _, CA := range mergedData {
+		caData[CA.File] = string(CA.Data)
+	}
+
+	err = optr.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Delete(context.TODO(), "merged-trusted-image-registry-ca", metav1.DeleteOptions{})
```

Comment:
wire a proper context in

Reply:
@jkyros WDYT? I know thew MCO uses context.TODO practically everywhere for large operations like this

Reply:
@codern that is because we are...very deficient with our context handling. Can I plead the 5th?   :stuck_out_tongue: 



This is potentially beyond the scope of this PR to immediately rectify though, we'd have to plumb that main context all the way through to your context: https://github.com/openshift/machine-config-operator/blob/be6c3023c9d292dd8df8abbfb57fa57d54639789/cmd/machine-config-operator/start.go#L68-L72

Reply:
https://github.com/openshift/machine-config-operator/pull/3194  is stale now, but we really should do something like it eventually